RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+  content: fileBlob
+});
+
+// Pause & resume upload
+let currentTransferStatus = uploadTask.pause();
+currentTransferStatus = uploadTask.resume();
+
+// Get the current progress of the upload
+const currentTransferProgress = uploadTask.getProgress();
+
+// Wait for the upload to finish (or fail)
+const uploadedObjectReference = await uploadTask.result;
+```
+
+Try out the new `storage` types here: https://www.typescriptlang.org/play#gist/292f4e24178bfca5881aa20961b930dc
+
+# `API` Category Changes
+Amplify is proposing the following changes for the `API` category.
+
+## First param is an object with named parameters
+To improve the readability of our APIs we will be introducing an object parameter to capture request parameters.
+**Amplify v5 (`aws-amplify@5`)**
+
+```typescript
+const apiName = 'MyApiName';
+const path = '/path';
+const myInit = {
+  headers: {}, // OPTIONAL
+  response: true, // OPTIONAL (return the entire Axios response object instead of only response.data)
+  queryStringParameters: {
+    name: 'param' // OPTIONAL
+  }
+};
+
+API.get(apiName, path, myInit)
+  .then((response) => {
+    // Add your code here
+  })
+  .catch((error) => {
+    console.log(error.response);
+  });
+```
+**Proposed Amplify v6 (`aws-amplify@6`)**
+```typescript
+API.get({
+  apiName: 'MyApi',
+  path: '/items',
+  authMode: 'AWS_IAM'
+}, myInit)
+  .then((result) => {
+    // Add your code here
+  }).catch((error) => {
+    console.log(error.response);
+  });
+```
+
+## Adding TypeScript generics to request body and response
+To improve developer experience and permit more strict typing we will be adding generic support to our `API` category APIs.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+Amplify v5 does not support using generics for the request body or response.
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```typescript
+type MyApiResponse = { firstName: string, lastName: string };
+
+API.get<MyApiResponse>({
+    apiName: 'MyApi',
+    path: '/getName'
+}).then(result => {
+    console.log(`The name is ${result.body.firstName} ${result.body.lastName}`)
+});
+
+API.put<string, { data: Array<number> }>({
+    apiName: '',
+    path: '/',
+    authMode: "API_KEY"
+}, {
+    headers: {
+        "Content-type": "text/plain",
+    },
+    body: "this is my content"
+}).then((result) => {
+    result.body.data.forEach((value) => console.log(value));
+});
+```
+
+## GraphQL operations have been split into query, mutation, and subscription
+To better capture customer intent and simplify API types we will split up the `graphql` API into individual APIs for queries, mutations, and subscriptions.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```typescript
+import { API } from "aws-amplify";
+import * as mutations from './graphql/mutations';
+import { GraphQLQuery, GraphQLSubscription } from '@aws-amplify/api';
+import { CreateTodoInput, CreateTodoMutation, OnCreateTodoSubscription } from './API';
+import * as subscriptions from './graphql/subscriptions';
+
```

Comment:
IMO we shouldn't include imports in the same code, I think it distracts focus away from the types & APIs.